src/components/ConsoleIO/ConsoleIO.test.tsx

```diff
@@ -0,0 +1,68 @@
+import { describe, it, expect, vi } from 'vitest';
+
+import { render, screen, fireEvent } from '../../test/test-utils';
+
+import { ConsoleIO } from './ConsoleIO';
+
+describe('ConsoleIO', () => {
+  it('renders tabs and sample area initially, then custom area when tab clicked', () => {
+    render(<ConsoleIO sampleValue="s" customValue="c" data-testid="consoleio-root" />);
+
+    const root = screen.getByTestId('consoleio-root');
```

Comment:
Tests rely on many getByTestId queries (example here) rather than semantic queries. Also the file uses fireEvent to change value which may not simulate user typing fully. Prefer semantic queries and userEvent for input interactions.

**Sugerencia:**
```typescript
Use `getByRole('textbox', { name: /sample-input/i })` or `getByLabelText` for textareas and `userEvent.setup()` + `user.type(element, 'text')` instead of `fireEvent.change` to better simulate real user interactions. Update tests to use semantic queries and userEvent for typing.
```